skrifa/src/color/color.rs

```diff
@@ -0,0 +1,259 @@
+use read_fonts::{
+    tables::colr::{ColrInstance, CompositeMode},
+    types::{BoundingBox, GlyphId, Point},
+    ReadError, TableProvider,
+};
+
+use std::{collections::HashSet, fmt::Debug};
+
+use super::{
+    colrv1_traversal::{get_clipbox_font_units, traverse_with_callbacks},
+    transforms::Transform,
+};
+
+use crate::prelude::LocationRef;
+
+#[derive(Clone, Debug, Default)]
+// This repr(C) is required so that C-side FFI's
+// are able to cast the ColorStop slice to a C-side array pointer.
+#[repr(C)]
+pub struct ColorStop {
```

Comment:
Just a random thought: I was thinking about parameterizing this on the (palette_index, alpha) payload so the type can support both paletted and resolved colors. Not really important now but just considering the future and other potential consumers.